File: src/features/formazione/corsi/soa.ts
import { NUMERI } from "../numeri";
import type { Sezione } from "../tipi";

/** Dichiarazione di Applicabilità (ISO/IEC 27001). Metodo dal corso del committente. */
export const SOA: Sezione[] = [
  {
    id: "contesto-e-ambito",
    titolo: "Contesto, ambito e quadri normativi",
    minuti: 6,
    sommario: "Il campo di applicazione, i due ruoli dichiarati, e quali quadri entrano nella dichiarazione.",
    blocchi: [
      {
        tipo: "prosa",
        testo:
          "Il campo di applicazione del sistema — servizi, processi, sedi, tecnologie coperti — è la prima cosa che un auditor legge, e le esclusioni dal campo vanno motivate. Poi due dichiarazioni che decidono quali controlli entreranno: il ruolo nel trattamento dei dati personali e la posizione rispetto ai servizi cloud.",
      },
      {
        tipo: "prosa",
        testo: `I controlli della norma principale sono sempre in ambito. Gli altri quadri — in tutto sono **${NUMERI.quadriSoa}** — si attivano quando il contesto lo richiede, e i loro controlli entrano nella dichiarazione insieme agli altri. Il catalogo completo porta **${NUMERI.controlliSoa} controlli** su **${NUMERI.sezioniSoa} sezioni**.`,
      },
      {
        tipo: "avviso",
        tono: "nota",
        titolo: "I ruoli sono valori chiusi, non testo libero",
        testo:
          "«Nessun servizio cloud» è una risposta, e va distinta dall'assenza di risposta. Nel prototipo la coerenza si controllava cercando la parola «cloud» nel testo: l'avviso «hai dichiarato il cloud ma non hai attivato il quadro» compariva proprio a chi aveva dichiarato di **non** usarne, e non c'era modo di farlo sparire. Con valori chiusi il confronto è per valore, e l'esaustività la controlla il compilatore.",
      },
      {
        tipo: "avviso",
        tono: "attenzione",
        titolo: "Attivare un quadro aggiunge lavoro, non solo righe",
        testo:
          "I controlli che arrivano entrano tutti come applicabili, e ciascuno vorrà uno stato, una motivazione, un riferimento documentale e un responsabile. Si attiva perché il contesto lo richiede, non per completezza.",
      },
    ],
  },

  {
    id: "i-controlli",
    titolo: "I controlli",
    minuti: 7,
    sommario: "Che cosa si dichiara per ognuno, e le quattro regole di compilazione.",
    blocchi: [
      {
        tipo: "tabella",
        intestazioni: ["Elemento", "Che cosa contiene"],
        righe: [
          [
            "Applicabile o escluso",
            "L'esclusione apre la giustificazione, che è obbligatoria: una dichiarazione che esclude un controllo senza dire perché è una non conformità già scritta",
          ],
          [
            "Stato di attuazione",
            "Dal non attuato all'attuato e verificato, passando per pianificato e parzialmente attuato",
          ],
          [
            "Motivazione della selezione",
            `Perché quel controllo è dentro: valutazione del rischio, obbligo legale, obbligo contrattuale, requisito di business, buona prassi. Sono **${NUMERI.motivazioniSoa}**, e se ne possono accendere più di una`,
          ],
          [
            "Riferimento documentale",
            "Il documento che dimostra il controllo. È quello che l'auditor chiederà di aprire",
          ],
          ["Responsabile", "Il presidio nominato, che diventa il responsabile predefinito nel piano"],
        ],
      },
      {
        tipo: "interfaccia",
        titolo: "Una sezione del registro dei controlli",
        vista: {
          genere: "stati",
          voci: [
            { testo: "5.1 · Politiche per la sicurezza delle informazioni", stato: "ok" },
            { testo: "5.7 · Threat intelligence", stato: "parziale" },
            { testo: "8.12 · Prevenzione della fuga di dati", stato: "no" },
            { testo: "7.4 · Sorveglianza della sicurezza fisica", stato: "na" },
          ],
        },
        nota: "L'ultimo è escluso, e un'esclusione senza giustificazione è una non conformità già scritta. Il terzo è applicabile e non attuato: pesa zero, non viene ignorato.",
      },
      {
        tipo: "avviso",
        tono: "nota",
        titolo: "Le motivazioni sono un insieme, non una scelta",
        testo:
          "Accenderne o spegnerne una è un'operazione atomica sull'elenco di quel controllo, non una riscrittura della riga. È la stessa ragione per cui in tutto il prodotto non si rimanda mai la riga intera: due modifiche ravvicinate si cancellerebbero a vicenda.",
      },
      {
        tipo: "elenco",
        voci: [
          "«Attuato» solo con un'evidenza reperibile: è ciò che verrà chiesto di vedere.",
          "«Attuato e verificato» quando un audit interno o un test ne ha confermato l'efficacia. Non è un grado di soddisfazione: è un fatto avvenuto.",
          "Almeno una motivazione per ogni applicabile. La valutazione del rischio è quella attesa dalla norma; le altre la rafforzano, non la sostituiscono.",
          "Si esclude solo ciò che è oggettivamente estraneo, e mai per comodità: l'esclusione è la parte della dichiarazione che viene letta con più attenzione.",
        ],
      },
      {
        tipo: "prosa",
        testo: `L'ordine che funziona: prima applicabilità ed esclusioni su tutto l'ambito, sezione per sezione; poi stato e motivazioni partendo dai **${NUMERI.cardineSoa} controlli cardine**; infine riferimenti documentali e responsabili. Le verifiche di coerenza dicono che cosa resta.`,
      },
    ],
  },

  {
    id: "calcolo-soa",
    titolo: "Come si calcola l'indice",
    minuti: 6,
    sommario: "Stati pesati, media sugli applicabili, fasce, e dove conviene lavorare.",
    blocchi: [
      {
        tipo: "formula",
        testo: "indice = Σ valore dello stato ÷ numero dei controlli applicabili",
      },
      {
        tipo: "avviso",
        tono: "errore",
        titolo: "Un controllo applicabile senza stato pesa ZERO, non viene ignorato",
        testo:
          "È il punto che una reimplementazione «ragionevole» sbaglia sempre. Mediare sui soli controlli valutati farebbe salire l'indice man mano che si saltano quelli difficili: si arriverebbe al novanta per cento avendo dichiarato dieci controlli su centotrenta. È il contrario del vero, e su una dichiarazione che va a un organismo di certificazione è una bugia con un numero sopra.",
      },
      {
        tipo: "prosa",
        testo: `Gli esclusi non contano nel denominatore — sono fuori dal sistema per una ragione dichiarata — e le fasce di maturità sono **${NUMERI.fasceSoa}**, dal non presidiato al pronto per la certificazione. Lo stesso calcolo produce il punteggio per quadro e per sezione.`,
      },
      {
        tipo: "formula",
        testo: "impatto di un'azione = (valore massimo − valore attuale) ÷ numero degli applicabili",
      },
      {
        tipo: "prosa",
        testo:
          "Per salire in fretta contano i molti controlli fermi, non l'uno portato alla perfezione: un controllo non attuato vale molto più di uno già parziale. È l'ordine con cui il piano propone le azioni.",
      },
      {
        tipo: "avviso",
        tono: "nota",
        titolo: "L'indice non è un requisito della norma",
        testo:
          "È lo strumento con cui il consulente governa il percorso, e l'obiettivo lo fissa l'organizzazione. La norma non chiede una percentuale: chiede che ogni controllo abbia una decisione motivata e, se selezionato, un'attuazione dimostrabile.",
      },
    ],
  },

  {
    id: "verifiche-soa",
    titolo: "Le verifiche di coerenza",
    minuti: 5,
    sommario: "Gli stessi controlli formali che un auditor esegue in apertura.",
    blocchi: [
      {
        tipo: "tabella",
        intestazioni: ["Rilievo", "Perché conta"],
        righe: [
          [
            "Esclusioni prive di giustificazione",
            "La dichiarazione deve motivare ogni controllo escluso: senza, l'organismo rileva una non conformità",
          ],
          [
            "Applicabili senza motivazione di inclusione",
            "Manca il perché: rischio, obbligo legale, contratto, business o buona prassi",
          ],
          [
            "Dichiarati attuati senza riferimento documentale",
            "Uno stato di attuazione va sostenuto da un documento o da una registrazione reperibile",
          ],
          ["Applicabili senza stato", "Finché resta vuoto, il controllo pesa zero sull'indice"],
          ["Applicabili senza responsabile", "Un controllo selezionato ha bisogno di un presidio nominato"],
          [
            "Profilo e quadri incoerenti",
            "Un ruolo o una posizione dichiarati senza il quadro corrispondente attivo, o il contrario",
          ],
        ],
      },
      {
        tipo: "prosa",
        testo:
          "Ogni rilievo elenca i codici dei controlli interessati: non è un punteggio, è una lista di posti dove andare. È la stessa forma della verifica negli altri percorsi, e per lo stesso motivo.",
      },
    ],
  },

  {
    id: "documento-soa",
    titolo: "Il documento",
    minuti: 4,
    sommario: "Che cosa contiene la Dichiarazione, e la nota che non si toglie.",
    blocchi: [
      {
        tipo: "elenco",
        voci: [
          "Una tabella per sezione con riferimento, controllo, applicabilità, motivazioni in sigla, stato e riferimento documentale.",
          "La legenda delle sigle, perché una tabella di sigle senza legenda si legge male proprio a chi la deve verificare.",
          "Il piano, con responsabili e termini.",
          "Le firme di chi ha redatto e di chi ha approvato.",
        ],
      },
      {
        tipo: "avviso",
        tono: "attenzione",
        titolo: "La nota di conformità al punto sulla Dichiarazione di Applicabilità",
        testo:
          "Sta riquadrata nel documento, in chiaro. È ciò che collega questa tabella all'obbligo della norma, e la sua assenza è il tipo di mancanza che si scopre in audit invece che prima.",
      },
      {
        tipo: "prosa",
        testo:
          "Il documento pubblicato è congelato: porta il codice di verifica e l'edizione dei contenuti con cui è stato costruito. Una revisione successiva è una versione nuova, non una modifica di quella consegnata — ed è ciò che permette di dire con certezza, sei mesi dopo, che cosa era stato dichiarato in quel momento.",
      },
    ],
  },
];

File: audio-formazione/distilla-slide/soa.py
# Slide della Dichiarazione di Applicabilità (ISO 27001).
from _scrivi import P, scrivi

SLIDE = {
    "contesto-e-ambito": [
        (0, "apertura", dict(
            titolo="Contesto e ambito", sottotitolo="Il confine, prima di qualunque controllo",
            testo="Servizi, processi, sedi e tecnologie coperti. Quello che resta fuori è escluso, con una motivazione scritta.",
            agenda=["Due dichiarazioni che decidono", "Valori chiusi", "I quadri", "Attivare per completezza",
                    "Concordare il confine", "Lo stesso confine ovunque"])),
        (1, "cards", dict(
            titolo="Due dichiarazioni che decidono i controlli",
            cards=[P("Ruolo nel trattamento", "Titolare o responsabile: obblighi diversi, controlli diversi."),
                   P("Servizi cloud", "Li usi, li offri, oppure non ne hai.")])),
        (2, "confronto", dict(
            titolo="«Non uso servizi cloud» è una risposta",
            a={"h": "Nel prototipo", "sub": "Si cercava la parola nel testo",
               "punti": ["L'avviso compariva a chi dichiarava di non usarne"]},
            b={"h": "Adesso", "sub": "Valori chiusi",
               "punti": ["Il confronto avviene per valore", "Quell'avviso non può più nascere"]})),
        (3, "numeri", dict(
            titolo="I quadri",
            numeri=[{"n": "Sempre", "h": "La norma principale", "d": "I suoi controlli non si scelgono."},
                    {"n": "{quadriSoa}", "h": "Quadri aggiuntivi", "d": "Si attivano quando il contesto lo richiede."},
                    {"n": "{controlliSoa}", "h": "Controlli, tutto acceso", "d": "Su {sezioniSoa} sezioni."}])),
        (4, "evidenza", dict(
            eyebrow="L'errore in buona fede",
            titolo="Un quadro si attiva perché serve, non per completezza.",
            testo="Ogni controllo che porta vuole stato, motivazione, riferimento e responsabile: decine di righe da compilare e da difendere.")),
        (5, "definizione", dict(
            eyebrow="La prima domanda del verificatore", titolo="Il campo descrive davvero l'azienda?",
            definizione="Scritto bene, il resto del documento si legge come una conseguenza. Scritto male, ogni controllo diventa discutibile.")),
        (6, "split", dict(
            titolo="Il confine è una decisione di governo",
            lead="Va concordato con chi in azienda ha l'autorità di dire di no.",
            punti=[P("Scritto da soli", "una sede o un servizio fuori dalle intenzioni della direzione"),
                   P("Una firma accanto", "dieci minuti adesso, e molti risparmiati dopo")])),
        (7, "chiusura", dict(
            titolo="Lo stesso confine ovunque",
            punti=[P("Nel manuale", "e nel contratto di certificazione"), P("Spesso sul sito", "dell'azienda"),
                   P("Confronta le parole", "non il senso")])),
    ],
    "i-controlli": [
        (0, "apertura", dict(
            titolo="I controlli", sottotitolo="Cinque dichiarazioni: due decidono, tre dimostrano",
            testo="Per ogni controllo del catalogo.",
            agenda=["Applicabile o escluso", "Lo stato di attuazione", "Le motivazioni", "Il riferimento documentale",
                    "Tre passate", "Che cosa chiede chi verifica"])),
        (1, "evidenza", dict(
            eyebrow="Prima decisione",
            titolo="Si esclude solo ciò che è estraneo, mai ciò che è difficile.",
            testo="L'esclusione vuole una giustificazione: senza, è una non conformità già scritta, e chi verifica comincia da lì.")),
        (2, "tabella", dict(
            titolo="Lo stato di attuazione",
            cols=["Stato", "Quando si dichiara"],
            righe=[["Non attuato", "Il controllo non c'è"], ["Pianificato", "È deciso, non ancora fatto"],
                   ["Parzialmente attuato", "C'è, ma non del tutto"], ["Attuato", "Con un'evidenza reperibile subito"],
                   ["Attuato e verificato", "Un audit o un test ne ha confermato l'efficacia, con una data"]])),
        (3, "split", dict(
            titolo="Le {motivazioniSoa} motivazioni della selezione",
            lead="Almeno una per controllo applicabile. La valutazione del rischio è quella attesa; le altre la rafforzano, non la sostituiscono.",
            punti=[P("Valutazione del rischio", "quella attesa dalla norma"), P("Obbligo legale", "o contrattuale"),
                   P("Requisito di business", "o buona prassi")])),
        (4, "definizione", dict(
            eyebrow="Un insieme, non una scelta unica", titolo="Motivazioni voce per voce",
            definizione="Accendere o spegnere una voce agisce solo su quella: in due sullo stesso controllo, nessuno cancella l'altro.")),
        (5, "confronto", dict(
            titolo="Il riferimento documentale",
            a={"h": "«Gestione degli accessi»", "sub": "Una descrizione", "punti": ["Sei mesi dopo non fa ritrovare niente"]},
            b={"h": "«Procedura 03.2 rev. 4»", "sub": "Il nome vero", "punti": ["Quello che l'auditor chiederà di aprire"]})),
        (7, "flusso", dict(
            titolo="Tre passate, non riga per riga",
            passi=[P("Prima", "applicabilità ed esclusioni, sezione per sezione"),
                   P("Seconda", "stato e motivazioni, dai {cardineSoa} controlli cardine"),
                   P("Terza", "riferimenti e responsabili: raccolta, delegabile")])),
        (8, "cards", dict(
            titolo="Le tre domande di chi verifica",
            cards=[P("Un controllo escluso", "E perché: la motivazione regge?"),
                   P("Un attuato scelto a caso", "E la sua evidenza: non bastano tre pronte."),
                   P("Chi se ne occupa", "A voce: un nome e un ruolo, non l'ufficio.")])),
        (9, "definizione", dict(
            eyebrow="Con il cliente", titolo="Non si valuta l'azienda",
            definizione="Si scrive che cosa ha deciso di presidiare e perché: un non attuato diventa una riga del piano, non una bocciatura.")),
        (10, "chiusura", dict(
            titolo="Rileggere le motivazioni",
            punti=[P("Una settimana dopo", "metà risultano generiche"),
                   P("Una motivazione che vale per tutti", "non motiva nessuno"),
                   P("È lì", "che si capisce se il lavoro è stato fatto o compilato")])),
    ],
    "calcolo-soa": [
        (0, "apertura", dict(
            titolo="Il calcolo", sottotitolo="L'indice di maturità",
            testo="Somma dei valori degli stati, divisa per i controlli applicabili: gli esclusi non entrano nel denominatore.",
            agenda=["Non valutato pesa zero", "Fasce e sezioni", "L'ordine del piano", "Non è un requisito",
                    "Due misurazioni", "Il punteggio per sezione"])),
        (1, "numeri", dict(
            titolo="Un controllo senza stato pesa zero",
            numeri=[{"n": "90%", "h": "Mediando solo i valutati", "d": "Con dieci controlli dichiarati su centotrenta."},
                    {"n": "Il vero", "h": "Pesando zero i non valutati", "d": "L'indice sale lavorando, non saltando."}])),
        (3, "confronto", dict(
            titolo="{fasceSoa} fasce, e il punteggio per sezione",
            a={"h": "Solo il totale", "sub": "Un 60% complessivo", "punti": ["Non dice dove intervenire"]},
            b={"h": "Per sezione", "sub": "Una al 20%, le altre all'80%", "punti": ["Dice da dove cominciare"]})),
        (4, "definizione", dict(
            eyebrow="La formula che ordina il piano", titolo="L'impatto di un'azione",
            definizione="(valore massimo − valore attuale) ÷ numero degli applicabili.",
            punti=[P("Venti controlli da zero a parziale", "spostano più di tre portati a verificato")])),
        (5, "evidenza", dict(
            eyebrow="Da dire prima di qualunque percentuale",
            titolo="L'indice non è un requisito della norma.",
            testo="La norma chiede decisioni motivate e attuazione dimostrabile. Al 70% con tutto motivato si è certificabili; al 90% con metà giustificazioni mancanti, no.")),
        (6, "split", dict(
            titolo="Guardarlo due volte",
            lead="Il numero assoluto dice poco; la differenza fra due misurazioni misura il lavoro fatto.",
            punti=[P("Salva il punteggio", "quando chiudi una passata"), P("Il termine di paragone", "è tuo, non di un settore")])),
        (7, "chiusura", dict(
            titolo="Il punteggio per sezione",
            punti=[P("Le sezioni", "non pesano uguale per ogni azienda"),
                   P("Servizi o manifattura", "nella stessa posizione numerica stanno diversamente"),
                   P("Un solo numero", "toglie l'informazione per decidere il budget")])),
    ],
    "verifiche-soa": [
        (0, "apertura", dict(
            titolo="Le verifiche di coerenza", sottotitolo="La prima mezz'ora dell'auditor",
            testo="Arrivare all'appuntamento avendo già trovato da solo quello che troverebbe lui.",
            agenda=["Sei verifiche", "Esclusioni e motivazioni", "Attuati senza documento", "Stati e responsabili",
                    "Profilo e quadri", "Presto e spesso"])),
        (1, "punti", dict(
            titolo="Sei verifiche, nell'ordine in cui arrivano",
            punti=[P("Esclusioni", "senza giustificazione"), P("Applicabili", "senza motivazione di inclusione"),
                   P("Attuati", "senza riferimento documentale"), P("Applicabili", "senza stato"),
                   P("Applicabili", "senza responsabile"), P("Profilo e quadri", "incoerenti")])),
        (2, "evidenza", dict(
            eyebrow="La prima",
            titolo="Un'esclusione senza motivazione non si rimedia in verifica.",
            testo="O la motivazione è scritta, o manca.")),
        (3, "definizione", dict(
            eyebrow="La seconda", titolo="Il perché che si rimanda",
            definizione="Rischio, obbligo legale, contratto, business o buona prassi: è il rilievo che si accumula più in fretta.")),
        (4, "evidenza", dict(
            eyebrow="La terza",
            titolo="Attuato senza documento: la contraddizione più visibile.",
            testo="È la più facile da cercare, e chi verifica la cerca subito.")),
        (5, "cards", dict(
            titolo="Quarta e quinta",
            cards=[P("Senza stato", "Il controllo pesa zero: ti dice dove l'indice è più basso di quanto potrebbe."),
                   P("Senza responsabile", "In verifica si chiede a voce chi se ne occupa: serve un nome.")])),
        (7, "definizione", dict(
            eyebrow="La sesta", titolo="Profilo e quadri incoerenti",
            definizione="Un ruolo o una posizione senza il quadro attivo, o il contrario: nasce dalle scelte del primo passo.")),
        (8, "confronto", dict(
            titolo="Un elenco di codici, non un punteggio",
            a={"h": "Un numero", "sub": "Ti dice come stai", "punti": ["E ti lascia dove sei"]},
            b={"h": "Un elenco di codici", "sub": "Ti dice che cosa aprire", "punti": ["Chiuso, il rilievo sparisce da solo"]})),
        (9, "numeri", dict(
            titolo="Presto e spesso",
            numeri=[{"n": "Alla fine", "h": "Decine di righe", "d": "Tutte insieme, con la scadenza addosso."},
                    {"n": "A ogni passata", "h": "Cinque o sei", "d": "Chiuse mentre ricordi il perché."}])),
        (10, "chiusura", dict(
            titolo="Con il cliente",
            punti=[P("Non sono errori tuoi", "sono le domande di fra un mese"), P("Trovate adesso", "che c'è tempo"),
                   P("E ha il vantaggio", "di essere vero")])),
    ],
    "documento-soa": [
        (0, "apertura", dict(
            titolo="La Dichiarazione di Applicabilità", sottotitolo="Il documento che consegni",
            testo="Quattro parti, e una nota che non si toglie.",
            agenda=["Quattro parti", "La nota di conformità", "Congelato alla pubblicazione", "Quando pubblicare", "Legenda e firme"])),
        (1, "cards", dict(
            titolo="Quattro parti",
            cards=[P("Una tabella per sezione", "Riferimento, controllo, applicabilità, motivazioni, stato, documento."),
                   P("La legenda delle sigle", "Per chi deve verificare: l'unico lettore per cui è scritto."),
                   P("Il piano", "Con responsabili e termini."),
                   P("Le firme", "Di chi redige e di chi approva: due persone.")])),
        (2, "evidenza", dict(
            eyebrow="Riquadrata, in chiaro",
            titolo="La nota di conformità collega la tabella all'obbligo.",
            testo="Senza, il documento sembra completo ed è solo un elenco. In una dichiarazione fatta da altri, cercala per prima.")),
        (3, "definizione", dict(
            eyebrow="Quando pubblichi", titolo="Il documento si congela",
            definizione="Con il codice di verifica e l'edizione dei contenuti: una revisione crea una versione nuova accanto alla precedente.")),
        (4, "confronto", dict(
            titolo="Sei mesi dopo, davanti all'organismo",
            a={"h": "«Ci ricordiamo che»", "sub": "Una ricostruzione", "punti": ["A memoria"]},
            b={"h": "«Qui c'è scritto»", "sub": "La versione di marzo", "punti": ["Aperta e mostrata"]})),
        (5, "split", dict(
            titolo="Pubblicare quando è pronto",
            lead="Non quando è il momento di consegnare.",
            punti=[P("A metà lavoro", "resta nell'archivio per sempre"),
                   P("Tre versioni ravvicinate", "e il cliente si chiede che cosa non andasse")])),
        (6, "chiusura", dict(
            titolo="Legenda e firme",
            punti=[P("Non hanno contenuto tecnico", "e si trascurano"), P("Sono le prime due", "che un organismo guarda"),
                   P("Mancanti", "rendono sospetto anche un documento buono")])),
    ],
}

scrivi("soa", SLIDE)
